scripts/update_scpca_checkpoints.py

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env python3
+
+# script to update checkpoint directories for scpca-nf for version 0.4
+# copies files to new checkpoint directory location and
+# adds scpca-metadata.json files as needed
+
+import argparse
+import json
+import subprocess
+
+import boto3
+import pandas
+
+### Functions ###
+def get_barcode(tech, barcode_dir):
+    """Get the barcode file appropriate for a 10X technology"""
+    barcode_dir = barcode_dir.rstrip("/") # remove trailing slash if needed
+    if "10Xv3" in tech:
+        barcode_file = f"{barcode_dir}/3M-february-2018.txt"
+    elif "10Xv2" in tech:
+        barcode_file = f"{barcode_dir}/737K-august-2016.txt"
+    else:
+        barcode_file = "NA"
+
+    return barcode_file
+
+
+def process_scrna(run, consts, overwrite):
+    """
+    Process a scrna run, moving internal checkpoint files from source to destination locations
+    and adding a scpca-meta.json file.
+    """
+    # skip if empty
+    if not isinstance(run.scpca_run_id, str):
+        return
+
+    print(f"Processing {run.scpca_run_id}")
+
+    # remove any slashes that might be at ends of bucket or prefix
+    bucket = consts.bucket.strip('/')
+    prefix = consts.prefix.strip('/')
+
+    origin_prefix = f"{prefix}/{consts.source_dir}/rad/{run.scpca_library_id}/{run.scpca_run_id}-rna"
+    dest_prefix = f"{prefix}/{consts.dest_dir}/rad/{run.scpca_library_id}/{run.scpca_run_id}-rna"
+    results_json = f"{prefix}/{consts.results_dir}/{run.scpca_project_id}/{run.scpca_sample_id}/{run.scpca_library_id}_metadata.json"
+    metadata_key = f"{dest_prefix}/scpca-meta.json"
+
+    publish_uri = f"s3://{bucket}/{prefix}/{consts.dest_dir}/rad/{run.scpca_library_id}"
+    rad_uri = f"{publish_uri}/{run.scpca_run_id}-rna"
+
+    # set up S3
+    s3 = boto3.resource('s3')
+    s3_bucket = s3.Bucket(bucket)
+
+    ### read output metadata, if present
+    try:
+        results_obj = s3_bucket.Object(results_json)
+        results_meta = json.loads(results_obj.get()['Body'].read().decode('utf-8'))
+    except s3.meta.client.exceptions.NoSuchKey:
+        # object did not exist
+        results_meta = {}
+    scpca_version = results_meta.get("workflow_version")
```

Comment:
Will this work if `results_meta` is empty?

Reply:
Yes, it will. `results_meta["workflow_version"]` won't, but `.get()` will return `Null` for a missing key, which is fine.